terraform/modules/service/variables.tf

```diff
@@ -0,0 +1,153 @@
+  type        = bool
+}
+
+
+variable "image" {
+  description = "The image used to start a container. This string is passed directly to the Docker daemon. By default, images in the Docker Hub registry are available. Other repositories are specified with either `repository-url/image:tag` or `repository-url/image@digest`"
+  type        = string
+  default     = null
+}
+
+variable "load_balancers" {
+  description = "Load balancer(s) for use by the AWS ECS service."
+  type = list(object({
+    target_group_arn = string
+    container_name   = string
+    container_port   = number
+  }))
+}
+
+variable "memory" {
+  description = "Amount (in MiB) of memory used by the task."
+  type        = number
+}
+
+variable "mount_points" {
+  description = "The mount points for data volumes in your container"
+  type = list(object({
+    containerPath = optional(string)
+    readOnly      = optional(bool)
+    sourceVolume  = optional(string)
+  }))
+  default = [
+    {
+      containerPath = "/mnt/efs"
+      sourceVolume  = "efs"
+      readOnly      = true
+    },
+    {
+      containerPath = "/tmp"
+      sourceVolume  = "tmp"
+      readOnly      = false
+    },
+    {
+      containerPath = "/var/log"
+      sourceVolume  = "var_log"
+      readOnly      = false
+    }
+  ]
+}
+
+
+variable "network_configurations" {
+  description = "Network configuration for the aws ecs service."
+  type = list(object({
+    subnets          = list(string)
+    assign_public_ip = string
+    security_groups  = list(string)
+  }))
+}
+
+variable "platform" {
+  description = "Object that describes standardized platform values."
+  type        = any
+}
+
+variable "port_mappings" {
+  description = "The list of port mappings for the container. Port mappings allow containers to access ports on the host container instance to send or receive traffic. For task definitions that use the awsvpc network mode, only specify the containerPort. The hostPort can be left blank or it must be the same value as the containerPort"
+  type = list(object({
+    appProtocol        = optional(string)
+    containerPort      = optional(number)
+    containerPortRange = optional(string)
+    hostPort           = optional(number)
+    name               = optional(string)
+    protocol           = optional(string)
+  }))
+  default = null
+}
+
+variable "service_name_override" {
+  default     = null
+  description = "Desired service name for the service tag on the aws ecs service.  Defaults to platform.service."
+  type        = string
+}
+
+variable "task_execution_role_arn" {
+  default     = null
+  description = "ARN of the task execution role that the Amazon ECS container agent and the Docker daemon can assume.  Defaults to creation of a new role."
+  type        = string
+}
+
+variable "task_app_role_arn" {
+  description = "ARN of IAM role that allows your Amazon ECS container task to make calls to other AWS services."
+}
```

Comment:
Why is this `task_app_role_arn` instead of `task_role_arn`? Also, type should be set to string.